src/saleor_app/app.py

```diff
@@ -1,39 +1,71 @@
-from typing import Awaitable, Callable
+import warnings
+from typing import Awaitable, Callable, Optional
 
-from fastapi import APIRouter, FastAPI
+from fastapi import APIRouter, Depends, FastAPI
+from fastapi.routing import APIRoute
 
+from saleor_app.deps import verify_saleor_domain, verify_webhook_signature
 from saleor_app.endpoints import handle_webhook, install, manifest
+from saleor_app.errors import ConfigurationError
+from saleor_app.http import WebhookRoute
 from saleor_app.schemas.core import DomainName, WebhookData
-from saleor_app.schemas.handlers import WebhookHandlers
+from saleor_app.schemas.handlers import SQSHandlers, WebhookHandlers
 from saleor_app.schemas.manifest import Manifest
+from saleor_app.settings import AWSSettings
 
 
 class SaleorApp(FastAPI):
     def __init__(
         self,
         *,
+        manifest: Manifest,
         validate_domain: Callable[[DomainName], Awaitable[bool]],
         save_app_data: Callable[[DomainName, WebhookData], Awaitable],
-        webhook_handlers: WebhookHandlers,
-        get_webhook_details: Callable[[DomainName], Awaitable[WebhookData]],
+        get_webhook_details: Optional[
+            Callable[[DomainName], Awaitable[WebhookData]]
+        ] = None,
+        http_webhook_handlers: Optional[WebhookHandlers] = None,
+        aws_settings: Optional[AWSSettings] = None,
+        sqs_handlers: Optional[SQSHandlers] = None,
+        use_insecure_saleor_http: bool = False,
+        development_auth_token: Optional[str] = None,
         **kwargs,
     ):
         super().__init__(**kwargs)
 
-        self.extra["saleor"] = {
-            "validate_domain": validate_domain,
-            "save_app_data": save_app_data,
-            "webhook_handlers": webhook_handlers,
-            "get_webhook_details": get_webhook_details,
-        }
+        self.manifest = manifest
+        self.http_webhook_handlers = http_webhook_handlers
+        self.sqs_handlers = sqs_handlers
+
+        if self.sqs_handlers:
```

Comment:
I guess `sqs_handlers` and `webhook_handlers` should have some common interface, even in terms of app setup, maybe like `def mount(self, app: FastAPI): ..` If google pub/sub shows up, we will have quite P O T E Z N Y constructor. with 3 `if's`